corehq/apps/data_dictionary/util.py

```diff
@@ -417,3 +417,15 @@ def get_used_props_by_case_type(domain):
                 props_by_case_type[case_type_bucket.key] = []
             props_by_case_type[case_type_bucket.key].append(prop_bucket.key)
     return props_by_case_type
+
+
+def get_case_property_group_name_for_properties(domain, case_type_name):
+    case_group_name_for_property = {}
+    for case_property_name, group_name in (
```

Comment:
nit: what do you think about pulling out the query construction which will be looped over into a separate variable, then using that variable in the for-loop statement? 

```suggestion
case_property_group_query = CaseProperty.objects.filter(
     case_type__name=case_type_name, case_type__domain=domain
 ).select_related('case_type').select_related('group').values_list('name', 'group__name')
 
 for case_property_name, group_name in case_property_group_query:
```

I bring this up because when I started reading the for-loop I was stuck trying to figure out what the query does whilst having to remember that I'm reading this in the context of a for-loop. It would be nice if I could mentally separate these two things.

Up to you.

Reply:
Totally valid, thanks for sharing the context as well. :+1: 

Include the change that you suggested.